docs/DeveloperGuide.md

```diff
@@ -351,89 +353,140 @@ Command: `note 1 n/Thoughts txt/Something`
 
 
 
-### \[Proposed\] Undo/redo feature
+### Undo/redo feature
+#### Implementation
+
+The undo/redo mechanism is facilitated by `HistoryManager`. `HistoryManager` manages the current model state as well as 
+the states that can be undone/redone. It does so by storing `State` objects. Each `State` object contains a 
+`ReadOnlyLibrary`, `ReadOnlyUserPrefs` and a `Predicate` used to decide which books should be visible to the user. 
+
+* `HistoryManager#addNewState()` — Add a new state to be used as the current state
+* `HistoryManager#undo()` — Restores the most recent previous state from its history.
+* `HistoryManager#redo()` — Restores the most recently undone state from its history.
 
-#### Proposed Implementation
+The undo and redo operations are exposed in the `Model` interface as `Model#undo()` and `Model#redo()` respectively.
+Whenever the user enters a command (besides `undo/redo/help/exit/list/view/find`), the previous state
```

Comment:
Maybe it is clearer to state the commands will be managed like in UG instead of those that wont be?